### ISSUE TITLE
cuda-gdb/pytorch crash

### ISSUE THREAD

**Issue by andyljones** [pytorch/pytorch#24159]
## 🐛 Bug

When a minimal PyTorch extension is run in cuda-gdb, it crashes with this error: 
```
cuda-gdb/7.12/gdb/block.c:456: internal-error: set_block_compunit_symtab: Assertion `gb->compunit_symtab == NULL' failed.
```

## To Reproduce

I've put together a minimal Docker container capturing the bug [here](https://github.com/andyljones/cuda-gdb-bug). In particular, see [this command](https://github.com/andyljones/cuda-gdb-bug/blob/master/run.sh#L7), which calls [this script](https://github.com/andyljones/cuda-gdb-bug/blob/master/bad.cu).

## Expected behavior

The minimal container also has a contrasting [good.cu](https://github.com/andyljones/cuda-gdb-bug/blob/master/good.cu) script which does the same task, but doesn't crash in cuda-gdb.

## Environment
Run from inside the container:
```
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.2 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04) 7.4.0
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce RTX 2080 Ti
Nvidia driver version: 430.40
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.16.4
[pip] torch==1.2.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      243  
[conda] mkl_fft                   1.0.12           py37ha843d7b_0  
[conda] mkl_random                1.0.2            py37hd81dba3_0  
[conda] pytorch                   1.2.0           py3.7_cuda10.0.130_cudnn7.6.2_0    pytorch
```

## Additional context
I got to this example by starting with a full PyTorch C++ extension [a la this tutorial](https://pytorch.org/tutorials/advanced/cpp_extension.html) and pulling bits off until it was easy to read.

I can only find [two](https://github.com/NVIDIA/cuda-gdb/issues/5) [other](https://github.com/daniilidis-group/neural_renderer/issues/65) people reporting this bug, one from 6th June and one from 7th August, so I assume it's due to something recent.  @sakaia , @maphysart , if you could provide your specs so I can see if it's just some magic combination of libraries causing it, that'd be a lot of help. The script to generate something like the above is

```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

I tried rolling the minimal example back to PyTorch 1.1, but the shared libs have apparently changed since then and I gave up on trying to get it to build. 

Going by the line number reported in the bug, cuda-gdb's internals have changed a bit since [the repo](https://github.com/NVIDIA/cuda-gdb) was last updated 4yrs ago.

**Comment by ezyang** [comment]
@andyljones Just to check, have you also reported this bug to NVIDIA?

**Comment by andyljones** [comment]
Not so far - was discouraged by how dead the cuda-gdb repo is. Do you know what the best way to report this sort of stuff is?

**Comment by ezyang** [comment]
https://developer.nvidia.com/nvidia_bug/add

**Comment by andyljones** [comment]
Thanks. Submitted, bug id is #2676602.

**Comment by andyljones** [comment]
NVIDIA have fixed this in the 10.1.243 CUDA release.
